app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java

```diff
@@ -18,23 +20,24 @@
       R.layout.activity_request_state_declined
   };
 
-  private SelectShopperActivity.SelectShopperAdapter.ShopRequest request;
+  private OutgoingRequest request;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     request = getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY);
-    setContentView(LAYOUTS[request.status.ordinal()]);
+    setContentView(LAYOUTS[request.getStatus().ordinal()]);
     ButterKnife.inject(this);
   }
 
   @Optional
   @OnClick(R.id.request_state_cancel_button)
   public void onCancelRequest(View view) {
+    request.setStatus(BaseRequest.RequestStatus.CANCELED);
     // TODO: Cancel request in server
-    request.shopper = null;
+    OutgoingRequest newRequest = new OutgoingRequest(request.getShoppingList());
     Intent intent = new Intent(this, SelectShopperActivity.class);
-    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, request);
+    intent.putExtra(SelectShopperActivity.EXTRAS_REQUEST_KEY, newRequest);
```

Comment:
You don't really need a request object to select a shopper. You might need the shopping list.
We could save a global shopping list draft that gets clear when you pay for it. That'll save us from passing it all over the place, and it'll actually save between sessions.